src/esp/gfx/RenderTarget.cpp

```diff
@@ -76,22 +81,17 @@ struct RenderTarget::Impl {
 
 #ifdef ESP_BUILD_WITH_TRIANGLE_SENSOR
     triangleIdBuffer_.setStorage(Mn::GL::RenderbufferFormat::R32I, size);
+    triangleFrameBuffer_ = Mn::GL::Framebuffer{{{}, framebufferSize()}};
+    triangleFrameBuffer_.attachRenderbuffer(TriangleIdBuffer, triangleIdBuffer_)
+        .mapForDraw({{1, TriangleIdBuffer}});
 #endif
 
     framebuffer_ = Mn::GL::Framebuffer{{{}, size}};
     framebuffer_.attachRenderbuffer(RgbaBuffer, colorBuffer_)
         .attachRenderbuffer(ObjectIdBuffer, objectIdBuffer_)
-#ifdef ESP_BUILD_WITH_TRIANGLE_SENSOR
-        .attachRenderbuffer(TriangleIdBuffer, triangleIdBuffer_)
-#endif
         .attachTexture(Mn::GL::Framebuffer::BufferAttachment::Depth,
                        depthRenderTexture_, 0)
-        .mapForDraw({{0, RgbaBuffer},
-                     {1, ObjectIdBuffer},
-#ifdef ESP_BUILD_WITH_TRIANGLE_SENSOR
-                     {2, TriangleIdBuffer}
-#endif
-        });
+        .mapForDraw({{0, RgbaBuffer}, {1, ObjectIdBuffer}});
```

Comment:
I didn't read the full code yet, but this immediately stands out. (Sorry, Magnum API is a bit messy in this case.) The `mapForDraw()` affects global state, which basically means the second call wipes out the result of the first. The original code before this change was more correct I think.